src/models/tree/build_tree.cc

```diff
@@ -29,6 +29,45 @@
 namespace legateboost {
 namespace {
 
+class BinnedX {
+  // These are stored as int16_t to save space
+  // Indices are relative to the feature, not the entire histogram
+  // The maximum number of bins in legate-boost is 2048
+  legate::Buffer<int16_t> data;
+
+ public:
+  legate::Rect<3> shape;
+  legate::Buffer<int32_t> row_pointers;
+  int64_t num_features;
+  int64_t num_rows;
+  template <typename T>
+  BinnedX(legate::AccessorRO<T, 3> X,
+          legate::Rect<3> shape,
+          const SparseSplitProposals<T>& split_proposals)
+    : shape(shape),
+      row_pointers(legate::create_buffer<int32_t, 1>(split_proposals.row_pointers.size())),
+      num_features(std::max(shape.hi[1] - shape.lo[1] + 1, 0ll)),
+      num_rows(std::max(shape.hi[0] - shape.lo[0] + 1, 0ll))
+  {
+    data = legate::create_buffer<int16_t>(num_features * num_rows);
+    std::copy(split_proposals.row_pointers.begin(),
+              split_proposals.row_pointers.end(),
+              row_pointers.ptr(0));
+    for (int i = 0; i < num_rows; i++) {
+      for (int j = 0; j < num_features; j++) {
+        auto bin_idx = split_proposals.FindBin(X[{i, j, 0}], j);
```

Comment:
This is surprising me a bit, although I would expect it to crash if wrong?  The old code uses `x_value = X[{index_global, j, 0}];`.  Shouldn't this accessor also use iterate global index points?

The speedup is certainly great!

Reply:
Yes this is a mistake - tests shouldnt be passing :/

Reply:
I've discovered CPU tests are not running properly. GPUs needs to be set explicitly to 0 and running with LEGATE_TEST=1 forces it to run the tests distributed across CPU cores, which exposes the issue.